From 3173f55b7924f8ff310fad48aed892818809dc90 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:49 -0400
Subject: [ethernet] mlx4: Fix bad use of dev_id

Message-id: <a234ced77b2e1d5b8233b1018cbb4b13b2e78044.1410425016.git.avadai@redhat.com>
Patchwork-id: 93633
O-Subject: [Patch RHEL7.1 V1 053/197] net/mlx4_en: Fix bad use of dev_id
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Amir Vadai <amirv@mellanox.com>

dev_id should be set for multiple netdev's sharing the same MAC, which
is not the case here.

Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit ca9f9f703950e5cb300526549b4f1b0a6605a5c5)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/en_netdev.c b/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
index caa488a..2c0823b 100644
--- a/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
@@ -2341,7 +2341,6 @@ int mlx4_en_init_netdev(struct mlx4_en_dev *mdev, int port,
  netif_set_real_num_rx_queues(dev, prof->rx_ring_num);
 
  SET_NETDEV_DEV(dev, &mdev->dev->pdev->dev);
- dev->dev_id =  port - 1;
  dev->dev_port = port - 1;
 
  /*
-- 
1.7.1